### ISSUE TITLE
[MPS] `torch.nn.functional.batch_norm` crashes with LLVM ERROR for `bool` tensors

### ISSUE THREAD

**Issue by christianahmed905** [pytorch/pytorch#188243]
### 🐛 Describe the bug

# [MPS] `torch.nn.functional.batch_norm` crashes with LLVM ERROR for `bool` tensors

## Describe the bug
Calling `torch.nn.functional.batch_norm` on an MPS device with a `bool` tensor causes a fatal process crash due to an `LLVM ERROR: Failed to infer result type(s)`.

## To Reproduce
```python
import torch
import torch.nn.functional as F

input_tensor = torch.zeros((1, 16, 8, 8, 4), dtype=torch.bool, device='mps')
running_mean = torch.zeros(16, dtype=torch.float32, device='mps')
running_var = torch.ones(16, dtype=torch.float32, device='mps')
weight = None
bias = torch.zeros(16, dtype=torch.float32, device='mps')

# This causes a fatal LLVM ERROR crash (signal 6)
F.batch_norm(
    input_tensor, 
    running_mean=running_mean, 
    running_var=running_var, 
    weight=weight, 
    bias=bias, 
    training=False, 
    momentum=0.5, 
    eps=0.0
)
```

## Expected behavior
The function should cleanly reject `bool` inputs with a `RuntimeError` indicating that `bool` is an unsupported dtype for batch normalization, rather than causing a hard crash via LLVM.

## Environment
- PyTorch version: (Latest)
- OS: macOS
- Device: Apple Silicon (MPS)

### Versions

``` 
Collecting environment information...
PyTorch version: 2.12.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 26.5 (arm64)
GCC version: Could not collect
Clang version: 21.0.0 (clang-2100.1.1.101)
CMake version: Could not collect
Libc version: N/A

Python version: 3.14.5 (main, May 10 2026, 19:20:57) [Clang 22.1.3 ] (64-bit runtime)
Python platform: macOS-26.5-arm64-arm-64bit-Mach-O
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Apple M5

Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] torch==2.12.0
[conda] Could not collect
```

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk @Isalia20

**Comment by malfet** [comment]
Closing as fixed by #188265
